src/rgw/rgw_lc.cc

```diff
@@ -959,9 +1015,26 @@ int RGWLC::bucket_lc_process(string& shard_id)
     return ret;
   }
 
-  ret = bucket_info.bucket.bucket_id.compare(bucket_id) ;
-  if (ret != 0) {
-    ldpp_dout(this, 0) << "LC:old bucket id found. " << bucket_name << " should be deleted" << dendl;
+  std::string_view cur_bucket_id = bucket_info.bucket.bucket_id;
+  if (cur_bucket_id != bucket_id) {
```

Comment:
since the `bucket_info.bucket.marker` stays the same across reshard (and not across delete/recreate), can we compare against that here instead? if the marker still matches, we know it's the same bucket and can avoid rewriting the bucket_id

Reply:
makes sense, we could avoid the whole writing the new ID thing, should make it easier for the admin tooling as well

Reply:
I think for newer buckets then we need to encode the bucket marker instead of the id, since cases where the LC policy was first applied after a reshard can't be handled otherwise.